### PR TITLE
v1.11 backport - Enable Google Analytics 4

### PR DESCRIPTION
**Pull request by qmonnet**
Backport of https://github.com/cilium/cilium/pull/22220, to enable Google Analytics on the v1.11 branch.

```release-note
docs: Enable Google Analytics for v1.11 documentation
```

Cc @chalin, @joestringer

We'll need a new docs-builder image. Joe, I haven't addressed the automation discussed on the recent PR for `master`, but at least I pushed to the `cilium` repository directly.

---

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 22220; do contrib/backporting/set-labels.py $pr done 1.11; done
```

### CODE REVIEW

**Review by nebril — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by qmonnet** [comment]
Ready to merge, pending TopHat's approval

**Pull request merged**
